src/sentry/tasks/relay.py

```diff
@@ -32,6 +32,23 @@ def update_config_cache(
     from sentry.relay import projectconfig_cache
     from sentry.relay.config import get_project_config
 
+    # At this point, we haven't marked the task as completed, so it should be
+    # debounced. If it isn't, another worker has picked a task for the same
+    # public key and marked it as done. Since we want to avoid recomputation,
+    # exit the current task.
+    if not projectconfig_debounce_cache.is_debounced(
```

Comment:
If we keep this addition (see @jjbayer's comment below): Instead of calling `is_debounced` here, could we have a higher-level method that:
- marks the task done if it's still scheduled and returns True
- returns False if that wasn't possible because the task was already marked as done.

That's not possible with Redis `DEL` command, but probably there are other simple ways to achieve this.